query/src/main/java/org/infinispan/query/clustered/QueryPartitioner.java

```diff
@@ -39,7 +39,7 @@ public QueryPartitioner(AdvancedCache<?, ?> cache) {
       localSegments.stream().forEach(bitSet::set);
       segmentsPerMember.put(localAddress, bitSet);
       for (int s = 0; s < numSegments; s++) {
-         if (!localSegments.contains(s)) {
+         if (!bitSet.get(s)) {
```

Comment:
We should probably fix line 39 as well. Should be `localSegments.forEach((IntConsumer) bitSet::set);` to avoid possible boxing and unboxing.